ops/framework.py

```diff
@@ -951,9 +951,9 @@ def _reemit(self, single_event_path: Optional[str] = None):
 
             else:
                 logger.warning(
-                    f'Reference to ops.Object at path {observer_path} has been garbage collected '
+                    'Reference to ops.Object at path %s has been garbage collected '
                     'between when the charm was initialised and when the event was emitted. '
-                    'Make sure sure you store a reference to the observer.'
+                    'Make sure sure you store a reference to the observer.', observer_path
```

Comment:
I'm surprised Ruff doesn't put `observer_path` on its own line.

Reply:
it would if there was a trailing comma.

or if the line was too long and had to be split.